selfservice/strategy/oidc/strategy.go

```diff
@@ -589,3 +589,39 @@ func (s *Strategy) CompletedAuthenticationMethod(ctx context.Context) session.Au
 		AAL:    identity.AuthenticatorAssuranceLevel1,
 	}
 }
+
+func (s *Strategy) processIDToken(w http.ResponseWriter, r *http.Request, provider Provider, idToken, idTokenNonce string) (*Claims, error) {
+	verifier, ok := provider.(IDTokenVerifier)
+	if !ok {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The provider %s does not support id_token verification", provider.Config().Provider))
+	}
+	claims, err := verifier.Verify(r.Context(), idToken)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Could not verify id_token").WithError(err.Error()))
+	}
+
+	if err := claims.Validate(); err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The id_token claims were invalid").WithError(err.Error()))
+	}
+
+	// First check if the JWT contains the nonce claim.
+	if claims.Nonce == "" {
+		// If it doesn't, check if the provider supports nonces.
+		if verifier.NonceSupported(claims) {
```

Comment:
We could also just require nonces to be present in the token regardless of the provider. If someone wants to integrate with an Apple device that does not support the nonce, we can still add this in later.

Reply:
Done! The check is a bit less readable now, if you have an idea on improvement, LMK.